boards/arm/apollo4p_evb/apollo4p_evb_defconfig

```diff
@@ -0,0 +1,10 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (c) 2023 Antmicro <www.antmicro.com>
+
+CONFIG_SOC_SERIES_APOLLO4X=y
+CONFIG_CONSOLE=y
+CONFIG_UART_CONSOLE=y
+CONFIG_SERIAL=y
+CONFIG_UART_INTERRUPT_DRIVEN=y
+CONFIG_HAS_AMBIQ_HAL=y
```

Comment:
why is board selecting HAL?

Reply:
I changed it so that this config is selected in SoC.

Reply:
an SoC doesn't have a HAL ;-)

Reply:
Can you explain what is the intended way to select this config? We use the function from the hal in [soc.c](https://github.com/zephyrproject-rtos/zephyr/pull/59470/files#diff-52af905f6fb03f5163df72c5aaae927e0e5bb3e2ad3f7e3d10f8f184cd23efdfR18-R20). In addition, each driver selects a different symbol provided for a specific driver, such as [stimer](https://github.com/zephyrproject-rtos/zephyr/pull/59470/files#diff-f72da9c776aca3dab9ea5c6b6659ce0101011ba230460023e0c1b3c27219a2edR11).